Data Structures/Heaps/DoubleEndedHeap.java

```diff
@@ -0,0 +1,213 @@
+
+public class DoubleEndedHeap {
+	int[] deap;
+	int elements = 0;
+
+	public DoubleEndedHeap() {
+		deap = new int[14];
+	}
+
+	// Returns true if index i is in max-heap, false otherwise.
+	private boolean inMaxHeap(int i) {
+		for (int j = i; j >= 2; j = (j - 1) / 2) {
+			if (j == 2) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	// Returns index of max partner when index pos is located in min-heap
+	private int maxPartner(int pos) {
+		Double exponent = Math.floor(Math.log(pos + 1) / Math.log(2)) - 1;
+		int temp = (int) (pos + Math.pow(2, exponent));
+		// Return the parent of the index location if there is no partner
+		if (temp > elements)
```

Comment:
Please use curly brackets. For example  
```java
if (temp > elements) {
   // something
}
```
see http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-142311.html#449